code/vote.c

```diff
@@ -132,11 +80,11 @@ bool sort_votes(char *hold[], int *holdi, int cabal)
 	}
 	else
 	{
-		return false;
+		retVal = false;
 	}
 
-	return true;
 	mysql_free_result(res);
+	return retVal;
```

Comment:
The way the function was written before `mysql_free_result` was never called. This was because the function returned before `mysql_free_result` was called. Moving `mysql_free_result` to before the return ensures it is called. The move also fixes a memory leak.

Reply:
woot woot patching those mem leaks!